jib-core/src/main/java/com/google/cloud/tools/jib/cache/DefaultCacheStorageWriter.java

```diff
@@ -303,4 +364,15 @@ private void writeSelector(DescriptorDigest selector, DescriptorDigest layerDige
       Files.move(temporarySelectorFile, selectorFile, StandardCopyOption.REPLACE_EXISTING);
     }
   }
+
+  private void writeMetadataFile(JsonTemplate template, Path destination) throws IOException {
+    Path temporaryFile = Files.createTempFile(null, null);
+    temporaryFile.toFile().deleteOnExit();
```

Comment:
Can you double-check this does not delete the renamed file after JVM terminates?

Reply:
This is the same logic we use to write the selectors, so it should be fine. In fact, I can probably replace the end of `writeSelector()` with this method (just realized I need to add the try-catch back in though)

Reply:
Sounds good.

> In fact, I can probably replace the end of `writeSelector()` with this method

Probably later, because we'll make lock-related changes.